relay-server/src/services/project.rs

```diff
@@ -1105,7 +1105,7 @@ impl Project {
         metric_outcomes: &MetricOutcomes,
         mut buckets: Vec<Bucket>,
     ) -> Option<(Scoping, ProjectMetrics)> {
-        let Some(project_state) = self.valid_state() else {
+        let Some(project_state) = self.state_value() else {
```

Comment:
This is an improvement, but the project might have already been physically removed from the project cache map at this point. IMO the ideal solution would be to prolong the project (state)'s lifetime, e.g. by `Arc`ing (see Slack discussion).